sunpy/time/time.py

```diff
@@ -188,8 +190,21 @@ def parse_time(time_string, time_format=''):
     else:
         # remove trailing zeros and the final dot to allow any
         # number of zeros. This solves issue #289
-        if '.' in time_string:
+        if '.' in time_string and '+' not in time_string and '-' not in time_string:                   
             time_string = time_string.rstrip("0").rstrip(".")
+
+        if '+' in time_string :
+            time_zone = time_string[time_string.rindex('+'):]
+            time_string_strip = time_string[:time_string.rindex('+')].rstrip("0").rstrip(".")
+            d = parse(time_string_strip + time_zone)
+            return d.astimezone(timezone('UTC'))
+        elif '-' in time_string :
+            time_zone = time_string[time_string.rindex('-'):]
+            time_string_strip = time_string[:time_string.rindex('-')].rstrip("0").rstrip(".")
+            d = parse(time_string_strip + time_zone)
+            return d.astimezone(timezone('UTC'))
```

Comment:
All this probably could be done using just one `if` and a regular expression like:
```python
time_zone = time_string[re.search('\+|\-', time_string).start():]
```
so we can avoid the code repetition.